ion-schema/src/isl/isl_type_reference.rs

```diff
@@ -131,6 +131,7 @@ impl IslTypeRefImpl {
         isl_version: IslVersion,
         value: &Element,
         inline_imported_types: &mut Vec<IslImportType>,
+        allow_variably_occurring_type: bool,
```

Comment:
This looks like it is creating more tech debt that will need to be dealt with when we get around to #157. What are your thoughts on fixing #157 now and then resubmitting this PR vs. keeping it as it?

Reply:
I will create another PR for `VariablyOccuringTypeReference`, until then keeping this PR open.